Community/bulk/bulk_action_host_example.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 BlueCat Networks (USA) Inc. and its affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from flask import make_response
+import io
+import csv
+from bluecat.api_exception import PortalException, APIException
+class bulk_handler:
+
+    output      = None
+    form_data   = None
+    errors      = 0
+    session     = None
+    config      = None
+    results     = {}
+
+    #These will be called in the order below
+
+    def __init__(self, g, form_data):
+        self.form_data = form_data
+        self.session = g
+        self.view = None
+        try:
+            self.config = g.user.get_api().get_entity_by_id(self.form_data.configuration.data)
+            self.view = self.config.get_view(self.form_data.view.data)
+        except PortalException:
+            self.errors = 1
+            g.user.logger.info("Unable to get configuration or view!")
+            self.config = None
+
+    def handle_row(self, key, row):
+        self.results[key] = row
+
+        if self.config is None:
+            return
+        try:
+            csv_record = row['fqdn']
+            csv_network = row['network']
+            csv_ip = row['ip']
+            csv_mac = row["mac"]
+            csv_ttl = row["ttl"]
+        except Exception as e:
+            self.errors = self.errors+1
+            self.results[key]['result'] = "Missing required value value: %s" % str(e)
+            return
+        try:
+            self.view.add_host_record(csv_record, [csv_ip], csv_ttl)
+        except (Exception, PortalException, APIException) as e:
+            self.errors = self.errors + 1
+            self.results[key]['result'] = "Unable to add host record. %s" %str(e)
+            return
+        self.results[key]['result'] = "Added %s" % csv_network
+        return
+
+
+
+    def go_no_go(self):
+        if self.errors > 0:
+            return True, "We had one or more errors"
```

Comment:
Should all the `if... else...` paths in the `go_no_go` method implementations return `True`? (Same for the method in the next file too)

Reply:
It's intended as a framework, some may want to return true, some false depending on the use case.